components/file_system/src/thread_io.rs

```diff
@@ -1,187 +1,218 @@
-            if line.is_empty() || !line.contains(' ') {
-                continue;
+            sentinel.bytes[io_type as usize].fetch_add(io_bytes - last_io_bytes, Ordering::Relaxed);
+            THREAD_IO_TOTAL[io_type as usize]
+                .fetch_add(io_bytes - last_io_bytes, Ordering::Relaxed);
+            private.last_bytes = io_bytes;
+            if let Some(new_io_type) = new_io_type {
+                private.io_type = new_io_type;
             }
-            let mut s = line.split_whitespace();
-
-            if let (Some(field), Some(value)) = (s.next(), s.next()) {
-                if let Ok(value) = u64::from_str(value) {
-                    match &field[..field.len() - 1] {
-                        "read_bytes" => io_bytes.read = value,
-                        "write_bytes" => io_bytes.write = value,
-                        _ => continue,
+            private_cell.set(private)
+        })
+    }
+
+    pub fn fetch_thread_io_bytes(_io_type: IOType) -> IOBytes {
+        fetch_exact_thread_io_bytes(&ThreadID::current())
+    }
+
+    fn fetch_exact_thread_io_bytes(id: &ThreadID) -> IOBytes {
+        let io_file_path = PathBuf::from("/proc")
+            .join(format!("{}", id.pid))
+            .join("task")
+            .join(format!("{}", id.tid))
+            .join("io");
+
+        if let Ok(io_file) = File::open(io_file_path) {
+            let reader = BufReader::new(io_file);
+            let mut io_bytes = IOBytes::default();
+
+            for line in reader.lines().flatten() {
+                if line.is_empty() || !line.contains(' ') {
+                    continue;
+                }
+                let mut s = line.split_whitespace();
+
+                if let (Some(field), Some(value)) = (s.next(), s.next()) {
+                    if let Ok(value) = u64::from_str(value) {
+                        match &field[..field.len() - 1] {
+                            "read_bytes" => io_bytes.read = value,
+                            "write_bytes" => io_bytes.write = value,
+                            _ => continue,
+                        }
                     }
                 }
             }
+
+            return io_bytes;
         }
 
-        return io_bytes;
+        IOBytes::default()
     }
 
-    IOBytes::default()
+    impl Default for ThreadIOPrivate {
+        fn default() -> ThreadIOPrivate {
+            ThreadIOPrivate {
+                io_type: IOType::Other,
+                last_bytes: IOBytes::default(),
+            }
+        }
+    }
 }
 
-impl Default for ThreadIOPrivate {
-    fn default() -> ThreadIOPrivate {
-        ThreadIOPrivate {
-            io_type: IOType::Other,
-            last_bytes: IOBytes::default(),
-        }
+#[cfg(target_os = "linux")]
+pub use self::linux::{fetch_all_thread_io_bytes, get_io_type, set_io_type};
+
+#[cfg(target_os = "linux")]
+use self::linux::fetch_thread_io_bytes;
+
+#[cfg(not(target_os = "linux"))]
+mod notlinux {
```

Comment:
It's not idiomatic to combine words as an identifier, so I prefer change the mod name to `non_linux`.